docs/csharp/tutorials/ranges-indexes.md

```diff
@@ -72,10 +72,13 @@ The following sample shows many of the reasons for those choices. Modify `x`, `y
 
 ## Type support for indices and ranges
 
-Indexes and ranges provide clear, concise syntax to access a single element or a subrange of elements in a sequence. An index expression typically returns the type of the elements of a sequence. A range expression typically returns the same sequence type as the source sequence.
+Indexes and ranges provide clear, concise syntax to access a single element or a range of elements in a sequence. An index expression typically returns the type of the elements of a sequence. A range expression typically returns the same sequence type as the source sequence.
 
 Any type that provides an [indexer](../programming-guide/indexers/index.md) with an <xref:System.Index> or <xref:System.Range> parameter explicitly supports indices or ranges respectively. An indexer that takes a single <xref:System.Range> parameter may return a different sequence type, such as <xref:System.Span%601?displayProperty=nameWithType>.
 
+> [!IMPORTANT]
+> The range operator often allocates a copy to store the range of elements returned. This causes extra allocations and copies, impacting performance.
```

Comment:
This makes it sound like you should avoid ranges altogether in performance sensitive code even though using them correctly with `Span` or `Memory` can improve performance. Maybe it could say something like this?

> Depending on the sequence type, the range operator can allocate a copy to store the range of elements returned. This causes extra allocations and copies, impacting performance. In performance sensitive code, consider using `Span<T>` or `Memory<T>` as the sequence type, since the range operator does not allocate for them.